### ISSUE TITLE
See the map from last activity

### ISSUE THREAD

**Issue by rwieners** [tderflinger/MMM-GConnect#5]
Hi,
it would be great to see the map of my last run within this module. Idk if the Garmin API is capable of this
![IMG_4832](https://github.com/user-attachments/assets/87739865-a17f-4772-8529-a8bbd0b96373)
I added a pic how it looks in the Garmin app.

**Comment by tderflinger** [comment]
Hello @rwieners, 
thanks for your idea contribution. I will take a closer look at the API and check out if the data is there. 
Best,
-Thomas

**Comment by tderflinger** [comment]
Hi @rwieners, quick update: Yes, the data is there in the .FIT and .TCX files. One thing with everytime downloading the respective .TCX or .FIT file would be that it might not be that good for the Raspberry Pi SD-card. MagicMirror normally runs on Raspberry Pis.

**Comment by rwieners** [comment]
yes, so if we want to use it, we have to set the sync rate pretty low I guess, like every few hours or twice a day. Not sure if anybody would accept this

**Comment by tderflinger** [comment]
@rwieners The latest update now includes a map view with the activity route. It uses MapLibre and MapTiler for the map. You need to get a free API key from MapTiler to use the map. The extension downloads the TCX file, converts it to a GeoJSON format and displays that on the map. Feel free to check it out and give feedback.
